"Could not load digger config" Error on PRs

Recently upgraded both the Digger backend and Workflow runner from version 0.4.6 to vLatest (0.6.20 at that time) and immediately started receiving this comment on all PRs post-merge:

`❌ Could not load digger config, error:
`
Checking the backend, this error appears to not be new, and was happening pre-update, but was potentially swallowed and not commented. This occurs on all PRs closed in the repo, even ones that do not affect any terraform code, so it is quite disturbing to developers who do not know why this message appears.

Notably, on the backend I see these errors corresponding to branches that the PR recently closed and deleted: 

```
PlainClone error: couldn't find remote ref "refs/heads/branch_name"
github.go:682: Error cloning and loading config: couldn't find remote ref "refs/heads/branch_name"
```

I assume this is some issue with the Digger backend attempting to check out the branch upon PR merge, but we automatically delete branches on PR merge, so they will always be deleted. Could this be related to https://github.com/diggerhq/digger/issues/1155?

--- a/backend/controllers/github.go
+++ b/backend/controllers/github.go
@@ -468,6 +468,7 @@ func handlePullRequestEvent(gh utils.GithubClientProvider, payload *github.PullR
 	isDraft := payload.PullRequest.GetDraft()
 	commitSha := payload.PullRequest.Head.GetSHA()
 	branch := payload.PullRequest.Head.GetRef()
+	action := *payload.Action
 
 	link, err := models.DB.GetGithubAppInstallationLink(installationId)
 	if err != nil {
@@ -476,13 +477,36 @@ func handlePullRequestEvent(gh utils.GithubClientProvider, payload *github.PullR
 	}
 	organisationId := link.OrganisationId
 
+	ghService, _, ghServiceErr := utils.GetGithubService(gh, installationId, repoFullName, repoOwner, repoName)
+	if ghServiceErr != nil {
+		log.Printf("GetGithubService error: %v", err)
+		return fmt.Errorf("error getting ghService to post error comment")
+	}
+
+	// here we check if pr was closed and automatic deletion is enabled, to avoid errors when
+	// pr is merged and the branch does not exist we handle that gracefully
+	if action == "closed" {
+		branchName, _, err := ghService.GetBranchName(prNumber)
+		if err != nil {
+			utils.InitCommentReporter(ghService, prNumber, fmt.Sprintf(":x: Could not retrive PR details, error: %v", err))
+			log.Printf("Could not retrive PR details error: %v", err)
+			return fmt.Errorf("Could not retrive PR details: %v", err)
+		}
+		branchExists, err := ghService.CheckBranchExists(branchName)
+		if err != nil {
+			utils.InitCommentReporter(ghService, prNumber, fmt.Sprintf(":x: Could not check if branch exists, error: %v", err))
+			log.Printf("Could not check if branch exists, error: %v", err)
+			return fmt.Errorf("Could not check if branch exists: %v", err)
+
+		}
+		if !branchExists {
+			log.Printf("automating branch deletion is configured, ignoring pr closed event")
+			return nil
+		}
+	}
+
 	diggerYmlStr, ghService, config, projectsGraph, _, _, err := getDiggerConfigForPR(gh, installationId, repoFullName, repoOwner, repoName, cloneURL, prNumber)
 	if err != nil {
-		ghService, _, err := utils.GetGithubService(gh, installationId, repoFullName, repoOwner, repoName)
-		if err != nil {
-			log.Printf("GetGithubService error: %v", err)
-			return fmt.Errorf("error getting ghService to post error comment")
-		}
 		utils.InitCommentReporter(ghService, prNumber, fmt.Sprintf(":x: Could not load digger config, error: %v", err))
 		log.Printf("getDiggerConfigForPR error: %v", err)
 		return fmt.Errorf("error getting digger config")

--- a/libs/ci/github/github.go
+++ b/libs/ci/github/github.go
@@ -334,6 +334,17 @@ func (svc GithubService) GetBranchName(prNumber int) (string, string, error) {
 	return pr.Head.GetRef(), pr.Head.GetSHA(), nil
 }
 
+func (svc GithubService) CheckBranchExists(branchName string) (bool, error) {
+	_, resp, err := svc.Client.Repositories.GetBranch(context.Background(), svc.Owner, svc.RepoName, branchName, 3)
+	if err != nil {
+		if resp != nil && resp.StatusCode == 404 {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
 func ConvertGithubPullRequestEventToJobs(payload *github.PullRequestEvent, impactedProjects []digger_config.Project, requestedProject *digger_config.Project, config digger_config.DiggerConfig) ([]scheduler.Job, bool, error) {
 	workflows := config.Workflows
 	jobs := make([]scheduler.Job, 0)

